ibm/service/database/resource_ibm_database.go

```diff
@@ -1561,35 +1584,39 @@ func resourceIBMDatabaseInstanceCreate(context context.Context, d *schema.Resour
 
 		for _, user := range userList.(*schema.Set).List() {
 			userEl := user.(map[string]interface{})
-			createDatabaseUserRequestUserModel := &clouddatabasesv5.User{
-				Username: core.StringPtr(userEl["name"].(string)),
-				Password: core.StringPtr(userEl["password"].(string)),
-			}
+			if userEl["name"].(string) == "repl" && (strings.Contains(serviceName, "postgresql") || strings.Contains(serviceName, "enterprisedb")) {
```

Comment:
Instead of a special case for the `repl` user, we should attempt to update the password if user creation fails in case it already exists i.e. restoring from a backup